hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/compact/HoodieFlinkMergeOnReadTableCompactor.java

```diff
@@ -69,35 +67,22 @@ public void preCompact(
     }
   }
 
-  public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
-                                   HoodieWriteConfig writeConfig,
+  public List<WriteStatus> compact(HoodieWriteConfig writeConfig,
                                    CompactionOperation operation,
                                    String instantTime,
                                    TaskContextSupplier taskContextSupplier,
-                                   Option<HoodieReaderContext<?>> readerContextOpt,
+                                   HoodieReaderContext<?> readerContext,
                                    HoodieTable table) throws IOException {
     String maxInstantTime = getMaxInstantTime(table.getMetaClient());
```

Comment:
There is a read config `hoodie.file.group.reader.enabled`, it looks like we force to use FG reader for compaction now? Are we planning to remove this option.

Reply:
Yes, the plan is to have all reads go through the FileGroupReader as our common read path as part of the 1.1.0 release